master/buildbot/steps/shell.py

```diff
@@ -112,7 +112,7 @@ def __init__(self, workdir=None,
         buildstep_kwargs = {}
         # workdir is here first positional argument, but it belongs to BuildStep parent
         kwargs['workdir'] = workdir
-        for k in kwargs.keys()[:]:
+        for k in list(kwargs)[:]:
```

Comment:
I can't remember what [:] does? It some python magic to return a copy of the list if memory serves me right. Since the list is modified during iteration, maybe it should stay?